sklearn/feature_selection/tests/test_base.py

```diff
@@ -1,12 +1,15 @@
 import numpy as np
 from scipy import sparse as sp
+from scipy.stats import pearsonr, spearmanr
```

Comment:
> Also, I am unable to use mutual_info_classif since the input to this has to be modified.

Do you mean that `X` has to contain discrete variables? Ideally, `SelectKBest` would be used towards the end of a scikit-learn [pipeline](http://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html). So you can assume that all of the preprocessing has been performed already.